### PR TITLE
MM-67974: Add disk space info to Support Packet for local file store

### PR DESCRIPTION
**Pull request by hanzei**
## Summary

```yml
file_store:
  file_status: OK
  file_driver: local
  total_mb: 866560
  available_mb: 571155
  filesystem_type: ext4
```

## AI Summary

- Adds `total_mb`, `available_mb`, and `filesystem_type` fields to the `file_store` section of `diagnostics.yaml` in the Support Packet
- Fields are populated only when `DriverName == "local"`; omitted (zero/empty) for S3 and unsupported platforms (Windows etc.)
- Uses `syscall.Statfs` on Linux and `unix.Statfs` on Darwin, wrapped in a goroutine with a 5-second timeout to guard against stale NFS mounts
- Filesystem type is mapped from magic numbers (Linux) or `Fstypename` (Darwin) — supports ext4, xfs, nfs, btrfs, tmpfs, smb, fuse, overlay

#### Ticket Link

Closes [MM-67974](https://mattermost.atlassian.net/browse/MM-67974)

#### Release Note
```release-note
Added `client.Audit` on `pluginapi.Client` for plugins to emit audit records via the server audit pipeline (server 10.10+).
```

[MM-67974]: https://mattermost.atlassian.net/browse/MM-67974?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Change Impact: 🟠 Medium

**Regression Risk:** Adds new optional fields to a public diagnostics struct and platform-specific filesystem probing that runs in goroutines with timeouts; while changes are mostly isolated to support packet generation, the added public fields change the diagnostics output and untested paths (statfs timeout and disk-error branch) present moderate regression risk.  
**QA Recommendation:** Moderate manual QA—verify support packet YAML still serializes correctly, confirm disk fields populate on Linux/Darwin with local file store and remain empty for S3/unsupported platforms, exercise the statfs timeout and error path to confirm errors are recorded (and the recent error-variable fix behaves correctly).

*Generated by CodeRabbitAI*

### CODE REVIEW

File: server/channels/app/platform/support_packet_test.go [hunk 1]
```diff
@@ -237,6 +237,15 @@ func TestGetSupportPacketDiagnostics(t *testing.T) {
...
 		assert.Equal(t, "local", d.FileStore.Driver)
+		if runtime.GOOS == "linux" || runtime.GOOS == "darwin" {
+			assert.Positive(t, d.FileStore.TotalMB, "TotalMB should be positive on supported platforms")
+			assert.Positive(t, d.FileStore.AvailableMB, "AvailableMB should be positive on supported platforms")
+			assert.NotEmpty(t, d.FileStore.FilesystemType, "FilesystemType should not be empty on supported platforms")
+		} else {
+			assert.Zero(t, d.FileStore.TotalMB)
+			assert.Zero(t, d.FileStore.AvailableMB)
+			assert.Empty(t, d.FileStore.FilesystemType)
+		}
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/channels/app/platform/support_packet_test.go:248]: Running the tests with a full file system is a edge case that is not worth fixing as the would degrade the impact of the test.

[the remark this answers is not in the document]
Comment (hanzei) [reply server/channels/app/platform/support_packet_test.go:248]: Running the tests with a full file system is a edge case that is not worth fixing as the would degrade the impact of the test.

**Review by vish9812 — APPROVED** [review]
